pkg/protocol/engine/mempool/conflictdag/tests/tests.go

```diff
@@ -0,0 +1,301 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/iotaledger/iota-core/pkg/core/acceptance"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine/mempool/conflictdag"
+)
+
+func TestAll[ConflictID, ResourceID conflictdag.IDType, VotePower conflictdag.VotePowerType[VotePower]](t *testing.T, frameworkProvider func(*testing.T) *Framework[ConflictID, ResourceID, VotePower]) {
+	for testName, testCase := range map[string]func(*testing.T, *Framework[ConflictID, ResourceID, VotePower]){
+		"CreateConflict":               CreateConflict[ConflictID, ResourceID, VotePower],
+		"TestJoinConflictSets":         TestJoinConflictSets[ConflictID, ResourceID, VotePower],
+		"UpdateConflictParents":        UpdateConflictParents[ConflictID, ResourceID, VotePower],
+		"LikedInstead":                 LikedInstead[ConflictID, ResourceID, VotePower],
+		"CreateConflictWithoutMembers": CreateConflictWithoutMembers[ConflictID, ResourceID, VotePower],
+		"ConflictAcceptance":           ConflictAcceptance[ConflictID, ResourceID, VotePower],
+		"CastVotes":                    CastVotes[ConflictID, ResourceID, VotePower],
+		"CastVotes_VotePower":          CastVotesVotePower[ConflictID, ResourceID, VotePower],
+		"CastVotesAcceptance":          CastVotesAcceptance[ConflictID, ResourceID, VotePower],
+	} {
+		t.Run(testName, func(t *testing.T) { testCase(t, frameworkProvider(t)) })
```

Comment:
Because that way you can have one set of tests that test behavior of the interface, regardless of the underlying implementation. 
It is still possible to run a single test, but you need to run all of them and then in the IDE you can select one and run it individually.